common/Helpers/RuntimeHelper.cs

```diff
@@ -35,6 +39,35 @@ public static bool IsCurrentProcessRunningAsAdmin()
         return identity.Owner?.IsWellKnown(WellKnownSidType.BuiltinAdministratorsSid) ?? false;
     }
 
+    // Determine whether the current process is running elevated in a split token session
+    // will not return true if UAC is disabled and the user is running as administrator by default
+    public static unsafe bool IsCurrentProcessRunningElevated()
+    {
+        HANDLE tokenHandle;
+        if (!PInvoke.OpenProcessToken(PInvoke.GetCurrentProcess(), TOKEN_ACCESS_MASK.TOKEN_QUERY, &tokenHandle))
+        {
+            throw new System.ComponentModel.Win32Exception(Marshal.GetLastWin32Error());
+        }
+
+        try
+        {
+            TOKEN_ELEVATION_TYPE elevationType;
+            uint elevationTypeSize = (uint)Unsafe.SizeOf<TOKEN_ELEVATION_TYPE>();
+            uint returnLength;
+
+            if (!PInvoke.GetTokenInformation(tokenHandle, TOKEN_INFORMATION_CLASS.TokenElevationType, &elevationType, elevationTypeSize, &returnLength))
+            {
+                throw new System.ComponentModel.Win32Exception(Marshal.GetLastWin32Error());

```

Comment:
Do we need to handle these exceptions?

Reply:
This should pretty much never throw, and if it does, we want to get a crash dump rather than continuing execution.